dotCMS/src/main/java/com/dotcms/storage/DataBaseStorage.java

```diff
@@ -0,0 +1,556 @@
+        final boolean isExternalPool = UtilMethods.isSet(jdbcPool);
+
+        return isExternalPool?
+                DbConnectionFactory.getConnection(jdbcPool):
+                DbConnectionFactory.getConnection();
+    }
+
+    protected void wrapCloseConnection (final VoidDelegate voidDelegate) {
+
+        final String  jdbcPool       = Config.getStringProperty("DATABASE_STORAGE_JDBC_POOL_NAME", StringPool.BLANK);
+        final boolean isExternalPool = UtilMethods.isSet(jdbcPool);
+
+        if(isExternalPool) {
+
+            wrapExternalCloseConnection(voidDelegate, jdbcPool);
+        } else {
+
+            wrapLocalCloseConnection(voidDelegate);
+        }
+    }
+
+    private void wrapLocalCloseConnection(final VoidDelegate voidDelegate) {
+
+        final boolean isNewConnection = !DbConnectionFactory.connectionExists();
+
+        try {
+
+            voidDelegate.execute();
+        }  catch (DotSecurityException | DotDataException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            throw new DotRuntimeException(e);
+        } finally {
+
+            if (isNewConnection) {
+
+                DbConnectionFactory.closeSilently();
+            }
+        }
+    }
+
+    private void wrapExternalCloseConnection(final VoidDelegate voidDelegate, final String jdbcPool) {
+
+        Connection connection = null;
+        try {
+
+            connection = DbConnectionFactory.getConnection(jdbcPool);
+            voidDelegate.execute();
+        } catch (DotSecurityException | DotDataException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            throw new DotRuntimeException(e);
+        } finally {
+
+            CloseUtils.closeQuietly(connection);
+        }
+    }
+
+    protected  <T> T wrapInTransaction (final ReturnableDelegate<T> delegate) {
+
+        final String  jdbcPool       = Config.getStringProperty("DATABASE_STORAGE_JDBC_POOL_NAME", StringPool.BLANK);
+        final boolean isExternalPool = UtilMethods.isSet(jdbcPool);
+
+        if(isExternalPool) {
+
+            return this.wrapInExternalTransaction(delegate, jdbcPool);
+        } else {
+
+            try {
+
+                return LocalTransaction.wrapReturnWithListeners(delegate);
+            } catch (Exception e) {
+
+                Logger.error(this, e.getMessage(), e);
+                throw new DotRuntimeException(e);
+            }
+        }
+    }
+
+    private <T> T wrapInExternalTransaction(final ReturnableDelegate<T> delegate, final String jdbcPool) {
+
+        T result = null;
+        Connection connection = null;
+        boolean    autocommit = false;
+
+        try {
+
+            connection = DbConnectionFactory.getConnection(jdbcPool);
+            autocommit = connection.getAutoCommit();
+            connection.setAutoCommit(false);
+
+            result = delegate.execute();
+
+            connection.commit();
+        } catch (Throwable e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A catch statement should never catch throwable since it includes errors.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)